src/v/utils/fragmented_vector.h

```diff
@@ -97,6 +98,17 @@ class fragmented_vector {
     }
     ~fragmented_vector() noexcept = default;
 
+    template<typename Iter>
+    requires std::input_iterator<Iter>
+    fragmented_vector(Iter begin, Iter end)
+      : fragmented_vector() {
+        // Improvement: Write a more efficient implementation for
+        // random_access_iterators
+        for (auto it = begin; it != end; ++it) {
```

Comment:
isn't this just std::copy()

Reply:
Yes and no, we'd have to do it per fragment.

That however gets more complicated because you then have to calculate the chunks and even doing something like `std::distance`/`std::next` to get the end iterator for the chunk then actually is quite expensive for non-random iterators.

So we'd just need two overloads which isn't hard but just not needed at this point.